### PR TITLE
Add "ICU on WebAssembly" notes to globalization-icu.md

### PR DESCRIPTION
**Pull request by EgorBo**
## Summary

Addresses https://github.com/dotnet/runtime/issues/39285#issuecomment-674472316

cc @safern

### CODE REVIEW

File: docs/standard/globalization-localization/globalization-icu.md [hunk 1]
```diff
@@ -147,3 +147,14 @@ install_name_tool -change "libicuuc.67.dylib" "@loader_path/libicuuc.67.dylib" /
 ```
 LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name @loader_path/$(notdir $(MIDDLE_SO_TARGET))
 ```
+
+## ICU on WebAssembly
+In order to provide globalization compatibility with desktop profiles we aslo ship a special version of ICU for WebAssembly workloads. To make it as small as possible we had to introduce a couple of limitations, it allowed us to reduce ICU data file size from 24Mb down to 1.4Mb (~0.3Mb if compressed with Brotli). The following APIs are either not supported or supported with limitations:
```
Comment (safern) [inline docs/standard/globalization-localization/globalization-icu.md:152]:
```suggestion
In order to provide globalization compatibility with desktop profiles we also ship a special version of ICU for WebAssembly workloads. To make it as small as possible we had to introduce a couple of limitations, that allowed us to reduce ICU data file size from 24Mb down to 1.4Mb (~0.3Mb if compressed with Brotli). The following APIs are either not supported or supported with limitations:
```

**Review by safern — COMMENTED** [review]
cc: @gewarren

File: docs/standard/globalization-localization/globalization-icu.md [hunk 2]
```diff
@@ -147,3 +147,14 @@ install_name_tool -change "libicuuc.67.dylib" "@loader_path/libicuuc.67.dylib" /
 ```
 LD_SONAME = -Wl,-compatibility_version -Wl,$(SO_TARGET_VERSION_MAJOR) -Wl,-current_version -Wl,$(SO_TARGET_VERSION) -install_name @loader_path/$(notdir $(MIDDLE_SO_TARGET))
 ```
+
+## ICU on WebAssembly
+In order to provide globalization compatibility with desktop profiles we also ship a special version of ICU for WebAssembly workloads. To make it as small as possible we had to introduce a couple of limitations, that allowed us to reduce ICU data file size from 24Mb down to 1.4Mb (~0.3Mb if compressed with Brotli). The following APIs are either not supported or supported with limitations:
+
+- `CultureInfo.EnglishName`
```
Comment (gewarren) [inline docs/standard/globalization-localization/globalization-icu.md:154]:
```suggestion
- <xref:System.Globalization.CultureInfo.EnglishName?displayProperty=nameWithType>
```

File: docs/standard/globalization-localization/globalization-icu.md [hunk 3]
```diff
@@ -147,3 +147,14 @@ install_name_tool -change "libicuuc.67.dylib" "@loader_path/libicuuc.67.dylib" /
...
+- `CultureInfo.NativeName`
+- `DateTimeFormatInfo.NativeCalendarName`
+- `RegionInfo.NativeName`
+- `String.Normalize` and `String.IsNormalized` don't support rarely used `NormalizationForm.FormKC` and `NormalizationForm.FormKD` forms.
```
Comment (gewarren) [inline docs/standard/globalization-localization/globalization-icu.md:158]:
```suggestion

The following APIs are supported with limitations:

- <xref:System.String.Normalize(System.Text.NormalizationForm)?displayProperty=nameWithType> and <xref:System.String.IsNormalized(System.Text.NormalizationForm)?displayProperty=nameWithType> don't support the rarely used <xref:System.Text.NormalizationForm.FormKC?displayProperty=nameWithType> and <xref:System.Text.NormalizationForm.FormKD?displayProperty=nameWithType> forms.
```

File: docs/standard/globalization-localization/globalization-icu.md [hunk 4]
```diff
@@ -147,3 +147,14 @@ install_name_tool -change "libicuuc.67.dylib" "@loader_path/libicuuc.67.dylib" /
...
+- `RegionInfo.CurrencyNativeName` returns the same value as `CurrencyEnglishName`.
```
Comment (gewarren) [inline docs/standard/globalization-localization/globalization-icu.md:159]:
```suggestion
- <xref:System.Globalization.RegionInfo.CurrencyNativeName?displayProperty=nameWithType> returns the same value as <xref:System.Globalization.RegionInfo.CurrencyEnglishName?displayProperty=nameWithType>.
```

**Review by gewarren — APPROVED** [review]
I left some suggestions. Thanks for adding this.

**Comment by gewarren** [comment]
Thanks! BTW, if you commit suggestions from the "Files changed" tab, you can batch them into a single commit.

**Pull request merged**